db/version_set.h

```diff
@@ -938,6 +942,8 @@ class VersionSet {
   std::vector<FileMetaData*> obsolete_files_;
   std::vector<std::string> obsolete_manifests_;
 
+  std::unordered_map<uint64_t, bool> ssts_grabbed_for_purge_;
```

Comment:
what's the `bool` for? can this be an `unordered_set`?

Reply:
According to the comment in `DBImpl#PurgeObsoleteFiles` above the declaration of `sst_live_map` local variable:
```
// Now, convert live list to an unordered map, WITHOUT mutex held;
// set is slow.
```
The `bool` has no use here, at least for now.

Reply:
[2459f7e#diff-71a34e2e4e097b44064da7d3a2b68bd8R638](https://github.com/facebook/rocksdb/commit/2459f7ec4e62c3fff2c121e14a3d70f4c01379a7#diff-71a34e2e4e097b44064da7d3a2b68bd8R638) explicitly changes `unordered_set` to `unordered_map` for performance reason. You mean this change is a typo?

Reply:
Oh, got it, I assumed the value of the map is used but it isn't. Siying might be able to clarify the motivation of that change.

Either way, once we have vector or autovector in JobContext and VersionStorageInfo, it won't matter as much which representation is chosen.